test/smallsint.ml

```diff
@@ -0,0 +1,128 @@
+  assert (check 1 (add zero one));
+  assert (check (-128) (add max_int one));
+  assert (check (-65) (add max_int (of_int 64)));
+  ()
+
+(* Asserting that subtraction performs as expected *)
+let () =
+  let open Signed.Int8 in
+  let check expected res =
+    let equal : int -> int -> bool = Stdlib.( = ) in
+    equal expected (to_int res) in
+  assert (check (-1) (sub zero one));
+  assert (check 127 (sub min_int one));
+  assert (check 64 (sub min_int (of_int 64)));
+  ()
+
+(* Asserting that multiplication performs as expected *)
+let () =
+  let open Signed.Int8 in
+  let check expected res =
+    let equal : int -> int -> bool = Stdlib.( = ) in
+    equal expected (to_int res) in
+  assert (check (-1) (mul minus_one one));
+  assert (check (-128) (mul minus_one min_int));
+  assert (check (-2) (mul max_int (of_int 2)));
+  ()
+
+(* Asserting that logical right shifting performs as expected *)
+let () =
+  let open Signed.Int8 in
+  let check expected res =
+    let equal : int -> int -> bool = Stdlib.( = ) in
+    equal expected (to_int res) in
+  assert (check (-1) (shift_right_logical minus_one 0));
+  assert (check 127 (shift_right_logical minus_one 1));
+  ()
+
+(* Asserting that integer parsing performs as expected *)
+let () =
+  let open Signed.Int8 in
+  let check expected res =
+    let equal : int -> int -> bool = Stdlib.( = ) in
+    equal expected (to_int res) in
+  assert (check 1 (of_string "1"));
+  assert (check 127 (of_string "127"));
+  assert (check (-128) (of_string "-128"));
+  assert (try (ignore (of_string "128"); false) with Failure _ -> true);
+  assert (try (ignore (of_string "-129"); false) with Failure _ -> true);
+  assert (check (-1) (of_string "-1"));
+  assert (check (0b01) (of_string "0b01"));
+  assert (check (-0b01) (of_string "-0b01"));
+  assert (check (0xf) (of_string "0xf"));
+  assert (check (-0xf) (of_string "-0xf"));
+  assert (check (0x7f) (of_string "0x7f"));
+  assert (check (-0x80) (of_string "-0x80"));
+  assert (try (ignore (of_string "0x80"); false) with Failure _ -> true);
+  assert (try (ignore (of_string "-0x81"); false) with Failure _ -> true);
+  ()
+
+(* Asserting that integer printing performs as expected *)
+let () =
+  let open Signed.Int8 in
+  let check expected res =
+    let equal : string -> string -> bool = Stdlib.( = ) in
+    equal expected res in
+  assert (check "1" (to_string (of_int 1)));
+  assert (check "127" (to_string (of_int 127)));
+  assert (check "-128" (to_string (of_int 128)));       (* FIXME: This value seems strange? *)
+  assert (check "-128" (to_string (of_int (-128))));
+  assert (check "127" (to_string (of_int (-129))));     (* FIXME: This value seems strange? *)
+  assert (check "-1" (to_string (of_int (-1))));
+  assert (check "f" (to_hexstring (of_int (0xf))));
+  assert (check "7f" (to_hexstring max_int));
+  assert (check "80" (to_hexstring min_int));
+  assert (check "81" (to_hexstring (of_int (-0x7f))));  (* FIXME: This value seems strange? *)
```

Comment:
I found some of these results strange, but I guess this is more a result of `of_int` not doing any bounds checking? I can remove the FIXMEs if this is the expected behavior.